### PR TITLE
Solr 2.7.0: SolrQuery::setHighlightQuery / getHighlightQuery (hl.q)

### PR DESCRIPTION
**Pull request by omars44**
Added API changes from PECL Solr 2.7.0

### CODE REVIEW

**Comment by omars44** [comment]
@Girgias ping!

**Comment by Girgias** [comment]
> @Girgias ping!
> 
> I wonder why the entities file doesn't exist anymore

The entities file should be generated by configure, so you shouldn't have it included in the git sources.

File: reference/solr/solrquery/gethighlightquery.xml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ <refnamediv>
+  <refname>SolrQuery::getHighlightQuery</refname>
+  <refpurpose>return the highlightquery (hl.q)</refpurpose>
+ </refnamediv>
+
+ <refsect1 role="description">
+  &reftitle.description;
+  <methodsynopsis>
+   <modifier>public</modifier> <type>string</type><methodname>SolrQuery::getHighlightQuery</methodname>
+   <void />
```
Comment (Girgias) [inline reference/solr/solrquery/gethighlightquery.xml:14]: Nit:
```suggestion
   <void/>
```

I don't know if you have stubs for the extension, but then you can partially use the gen_stub script to generate the docs.

File: reference/solr/solrquery/gethighlightquery.xml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  </methodsynopsis>
+  <para>
+  Returns previously set highlight query. This parameter allows you to highlight different terms or fields than those being used to retrieve documents.
+  </para>
```
Comment (Girgias) [inline reference/solr/solrquery/gethighlightquery.xml:18]: Nit: indentation but line break
```suggestion
  <para>
   Returns previously set highlight query.
   This parameter allows you to highlight different terms or fields than those being used to retrieve documents.
  </para>
```

File: reference/solr/solrquery/gethighlightquery.xml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+ <refsect1 role="returnvalues">
+  &reftitle.returnvalues;
+  <para>
+   Returns the current SolrHighlightQuery string, or null if not set.
+  </para>
+ </refsect1>
+ 
+</refentry>
+
+<!-- Keep this comment at the end of the file
```
Comment (Girgias) [inline reference/solr/solrquery/gethighlightquery.xml:31]: Nit:
```suggestion
</refentry>
<!-- Keep this comment at the end of the file
```

File: reference/solr/solrquery/sethighlightquery.xml [hunk 4]
```diff
@@ -0,0 +1,68 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- $Revision$ -->
+
+<refentry xml:id="solrquery.sethighlightquery" xmlns="http://docbook.org/ns/docbook" xmlns:xlink="http://www.w3.org/1999/xlink">
```
Comment (Girgias) [inline reference/solr/solrquery/sethighlightquery.xml:4]: Nit:
```suggestion
<?xml version="1.0" encoding="utf-8"?>
<refentry xml:id="solrquery.sethighlightquery" xmlns="http://docbook.org/ns/docbook" xmlns:xlink="http://www.w3.org/1999/xlink">
```

File: reference/solr/solrquery/gethighlightquery.xml [hunk 5]
```diff
@@ -0,0 +1,50 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- $Revision$ -->
+
+<refentry xml:id="solrquery.gethighlightquery" xmlns="http://docbook.org/ns/docbook" xmlns:xlink="http://www.w3.org/1999/xlink">
```
Comment (Girgias) [inline reference/solr/solrquery/gethighlightquery.xml:4]: Nit:
```suggestion
<?xml version="1.0" encoding="utf-8"?>
<refentry xml:id="solrquery.gethighlightquery" xmlns="http://docbook.org/ns/docbook" xmlns:xlink="http://www.w3.org/1999/xlink">
```

File: reference/solr/solrquery/sethighlightquery.xml [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+ <refsect1 role="description">
+  &reftitle.description;
+  <methodsynopsis>
+   <modifier>public</modifier> <type>SolrQuery</type><methodname>SolrQuery::setHighlightQuery</methodname>
+   <methodparam><type>string</type><parameter>q</parameter></methodparam>
+  </methodsynopsis>
+  <para>
+   A query to use for highlighting. This parameter allows you to highlight different terms or fields than those being used to retrieve documents.
+  </para>
```
Comment (Girgias) [inline reference/solr/solrquery/sethighlightquery.xml:18]: The manual's style guide says to not use personalization (i.e. using "you") could you rephrase the sentence?
```suggestion
  <para>
   A query to use for highlighting.
   This parameter allows you to highlight different terms or fields than those being used to retrieve documents.
  </para>
```

File: reference/solr/solrquery/sethighlightquery.xml [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+  <para>Default: value of q</para>
+  <para>SolrParam: hl.q</para>
+ </refsect1>
```
Comment (Girgias) [inline reference/solr/solrquery/sethighlightquery.xml:22]: I'm confused by those two para tags, what purpose to they serve?

File: reference/solr/solrquery/sethighlightquery.xml [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+     </listitem>
+    </varlistentry>
+   </variablelist>
+  </para>
+ </refsect1>
+
+ <refsect1 role="returnvalues">
+  &reftitle.returnvalues;
+  <para>
+   Returns the current SolrQuery object, if the return value is used.
```
Comment (Girgias) [inline reference/solr/solrquery/sethighlightquery.xml:43]: For automatic linking.
```suggestion
   Returns the current <classname>SolrQuery</classname> object, if the return value is used.
```

File: reference/solr/solrquery/sethighlightquery.xml [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+  </para>
+ </refsect1>
+ 
+</refentry>
+
+<!-- Keep this comment at the end of the file
```
Comment (Girgias) [inline reference/solr/solrquery/sethighlightquery.xml:49]: Nit:
```suggestion
</refentry>
<!-- Keep this comment at the end of the file
```

**Review by Girgias — CHANGES_REQUESTED** [review]
Sorry for the delay, I was somewhat busy

File: reference/solr/solrquery/sethighlightquery.xml (hunk 7 above)
Comment (omars44) [reply reference/solr/solrquery/sethighlightquery.xml:20]: extra info

File: reference/solr/solrquery/sethighlightquery.xml (hunk 6 above)
Comment (omars44) [reply reference/solr/solrquery/sethighlightquery.xml]: sure

**Comment by omars44** [comment]
@Girgias sorry took me a while to find this PR for some reason, I thought it was long time merged. Just tag me next time

**Review by Girgias — APPROVED** [review]

File: reference/solr/solrquery/gethighlightquery.xml [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+
+ </refsect1>
+
+ <refsect1 role="returnvalues">
+  &reftitle.returnvalues;
+  <para>
+   Returns the current SolrHighlightQuery string, or null if not set.
+  </para>
+ </refsect1>
+
```
Comment (Girgias) [inline reference/solr/solrquery/gethighlightquery.xml:27]:
```suggestion

```

File: reference/solr/solrquery/sethighlightquery.xml [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+ </refsect1>
+
+ <refsect1 role="returnvalues">
+  &reftitle.returnvalues;
+  <para>
+   Returns the current <classname>SolrQuery</classname> object, if the return value is used.
+  </para>
+ </refsect1>
+ 
+</refentry>
```
Comment (Girgias) [inline reference/solr/solrquery/sethighlightquery.xml:46]:
```suggestion
 </refsect1>

</refentry>
```
